### PR TITLE
Implement mania selection blueprint movements

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu/pull/3709
- [x] https://github.com/ppy/osu/pull/3708

### CODE REVIEW

**Comment by peppy** [comment]
Has issues as pointed out yesterday IRL (selection masks stick to bottom of screen, handle incorrectly if seeking while dragging). Will hold off further review for the time being.

**Comment by smoogipoo** [comment]
> handle incorrectly if seeking while dragging

Will leave this to be fixed after this PR and https://github.com/ppy/osu/pull/3711 is merged. Will fix in #3736, since the movement logic here is on a per-hitobject level whereas in #3736 they've been bulked together.

**Review by peppy — APPROVED** [review]

**Pull request merged**
